testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -423,6 +424,33 @@ public void testTrinoShowingSparkCreatedTables()
         onTrino().executeQuery("DROP TABLE " + trinoTableName(trinoTable));
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testPrestoWritingDataWithObjectStorageLocationProvider()
+    {
+        String baseTableName = "test_object_storage_location_provider";
+        String sparkTableName = sparkTableName(baseTableName);
+        String trinoTableName = trinoTableName(baseTableName);
+        String dataPath = "hdfs://hadoop-master:9000/user/hive/warehouse/test_object_storage_location_provider/obj-data";
+
+        String sparkTableDefinition = "CREATE TABLE %s (_string STRING, _bigint BIGINT) USING ICEBERG OPTIONS (" +
+                "'write.object-storage.enabled'=true," +
+                "'write.object-storage.path'='" + dataPath + "')";
+        onSpark().executeQuery(format(sparkTableDefinition, sparkTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('a_string', 1000000000000000)", trinoTableName));
+
+        Row result = row("a_string", 1000000000000000L);
+        QueryResult sparkSelect = onSpark().executeQuery(format("SELECT _string, _bigint FROM %s", sparkTableName));
+        assertThat(sparkSelect).containsOnly(result);
+
+        QueryResult prestoSelect = onTrino().executeQuery(format("SELECT _string, _bigint FROM %s", trinoTableName));
+        assertThat(prestoSelect).containsOnly(result);
+
+        QueryResult filesTableResult = onTrino().executeQuery(format("SELECT file_path FROM %s", trinoTableName("\"" + baseTableName + "$files\"")));
+        filesTableResult.rows().forEach(row -> assertTrue(((String) row.get(0)).contains(dataPath)));
+
+        onSpark().executeQuery("DROP TABLE " + sparkTableName);
```

Comment:
we should have two delete modes

- as is here: delete with spark
- another: delete with Trino + assertion that data files are being correctly removed, same way as it happens when storage location is not used